### ISSUE TITLE
Weird compiler inconsistency with String!

### ISSUE THREAD

**Issue by Crossroadsman** [Crossroadsman/ios10AndSwift3FromBeginnerToPaidProfessional#2]
Inconsistency noticed in commit: f67707e376b1e7de5fdac49a7d140d19929441e5

# Issue:
variable declared as String!, e.g., `var testString!` can be used in an interpolated string, e.g., `let newString = "Hello \(testString)"`. Option-clicking the `newString` declaration in Xcode will disclose that the type is String, not String? but when compiled, the string will print as "Hello Optional(<value in testString>)".

# Cause:
See SE-0054 and [this](http://stackoverflow.com/questions/39537177/swift-3-incorrect-string-interpolation-with-implicitly-unwrapped-strings)  Stack Overflow answer
Basically, this is caused by the elimination of implicitly unwrapped optionals as distinct types.

# Workaround:
Declare the variables as String instead of String!

**Comment by Crossroadsman** [comment]
Not a bug. Compiler reserves the right to interpret IUOs as ? or regular Non-optional at compile time. Behaviour may vary depending on compiler's decisions.
